services/datamanager/data_manager.go

```diff
@@ -61,21 +51,68 @@ func FromDependencies(deps resource.Dependencies, name string) (Service, error)
 	return resource.FromDependencies[Service](deps, Named(name))
 }
 
-// DataCaptureConfigs specify a list of methods to capture on resources.
-type DataCaptureConfigs struct {
-	CaptureMethods []DataCaptureConfig `json:"capture_methods"`
+// AssociatedConfig specify a list of methods to capture on resources and implements the resource.AssociatedConfig interface.
+type AssociatedConfig struct {
+	CaptureMethods []*DataCaptureConfig `json:"capture_methods"`
+}
+
+func newAssociatedConfig(attributes utils.AttributeMap) (*AssociatedConfig, error) {
+	md, err := json.Marshal(attributes)
+	if err != nil {
+		return nil, err
+	}
+	var conf AssociatedConfig
+	if err := json.Unmarshal(md, &conf); err != nil {
+		return nil, err
+	}
+	return &conf, nil
+}
+
+// Equals describes if an DataCaptureConfigs is equal to a given AssociatedConfig.
+func (ac *AssociatedConfig) Equals(other resource.AssociatedConfig) bool {
+	ac2, err := utils.AssertType[*AssociatedConfig](other)
+	if err != nil {
+		return false
+	}
+	if len(ac.CaptureMethods) != len(ac2.CaptureMethods) {
+		return false
+	}
+	for i := 0; i < len(ac.CaptureMethods); i++ {
+		if !ac.CaptureMethods[i].Equals(ac2.CaptureMethods[i]) {
+			return false
+		}
+	}
```

Comment:
should we order the capture methods?

Reply:
Like manually sort them before doing this check?  Not a bad idea, but from what I can tell they are ordered by default when deserializing from JSON so maybe not necessary

Reply:
You mean sort the capture methods and then compare for equality? Confused what you mean here.

Reply:
yep, cause otherwise if you change the order of the resources in the config, the config would have "changed". this isn't really a huge issue, but perhaps we can just leave a comment saying this would happen

Reply:
I will add a comment to this effect.